services/core-api/app/api/document_generation/resources/explosives_permit_document.py

```diff
@@ -0,0 +1,108 @@
+import requests
+from flask import request, Response, stream_with_context
+from flask_restplus import Resource, marshal
+from werkzeug.exceptions import BadRequest, InternalServerError, BadGateway
+from app.extensions import api, cache
+
+from app.api.utils.resources_mixins import UserMixin
+from app.api.constants import EXPLOSIVES_PERMIT_DOCUMENT_DOWNLOAD_TOKEN
+from app.config import Config
+
+from app.api.mines.documents.models.mine_document import MineDocument
+from app.api.mines.explosives_permit.models.explosives_permit import ExplosivesPermit
+from app.api.mines.explosives_permit.models.explosives_permit_document_type import ExplosivesPermitDocumentType
+from app.api.mines.explosives_permit.models.explosives_permit_document_xref import ExplosivesPermitDocumentXref
+
+from app.api.services.document_generator_service import DocumentGeneratorService
+from app.api.services.document_manager_service import DocumentManagerService
+from app.api.mines.explosives_permit.response_models import EXPLOSIVES_PERMIT_DOCUMENT_MODEL
+
+
+class ExplosivesPermitDocumentResource(Resource, UserMixin):
+    @api.doc(
+        description=
+        'Generates the document associated with the received token and pushes it to the Document Manager and associates it with the Explosives Permit. Returns either the file content or the document record, depending on query parameters.',
+        params={
+            'token':
+            'The UUID4 token used to generate the document. Must be retrieved from the Explosives Permit Document Type POST endpoint.',
+            'return_record':
+            'If true, returns the created document record, otherwise, returns the generated file content.',
+            'is_preview':
+            'If true, returns the generated document without creating the document record.'
+        })
+    def get(self):
+        token = request.args.get('token', '')
+        return_record = request.args.get('return_record') == 'true'
+        is_preview = request.args.get('is_preview') == 'true'
+        return ExplosivesPermitDocumentResource.generate_explosives_permit_document(
+            token, return_record, is_preview)
+
+    @classmethod
+    def generate_explosives_permit_document(cls,
+                                            token,
+                                            return_record,
+                                            is_preview=False,
+                                            commit=True):
+        # Ensure that the token is valid
+        token_data = cache.get(EXPLOSIVES_PERMIT_DOCUMENT_DOWNLOAD_TOKEN(token))
+        cache.delete(EXPLOSIVES_PERMIT_DOCUMENT_DOWNLOAD_TOKEN(token))
+        if not token_data:
+            raise BadRequest('Valid token required for download')
+
+        # Get the template associated with the token
+        document_type_code = token_data['document_type_code']
+        explosives_permit_document_type = ExplosivesPermitDocumentType.query.unbound_unsafe().get(
```

Comment:
Is there a reason this needs to use the unbound_unsafe query?

Reply:
This file is pretty much the same as the now_document.py generation resource in the same folder--I can't remember why exactly, but it might have to do with supporting being able to generate the document outside of a request.

Reply:
I also think we could probably do with a tech-debt ticket to "genericize" our document_generation resources. A lot of the code each one uses is largely the same with some minor differences.

Reply:
I agree